users.js

```diff
@@ -640,6 +641,19 @@ class User {
 		}
 		return true;
 	}
+	lockName() {
+		let num = '' + this.guestNum;
+		// Make sure the name is too long for regular name choosing.
+		while (num.length < 6) {
+			num = '0' + num;
+		}
+		let name = 'Pokemon Trainer #' + num;
+		this.forceRename(name, this.registered);
+		this.namelocked = true;
```

Comment:
You might want to set this before doing the rename; this allows config scripts to detect namelocks and helps with a few other things.